server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1097,6 +1074,45 @@ public Pool find(String poolId) {
     }
 
     /**
+     * Performs the work of the entitleByProducts method in its own transaction to help unlock
+     * pools which can no longer be bound.
+     * <p></p>
+     * This method should not be called directly, and is only declared protected to allow the
+     * @Transactional annotation to function.
+     *
+     * @param data
+     *  The autobind data to use for entitling a consumer
+     *
+     * @return
+     *  a list of entitlements created as for this autobind operation
+     */
+    @Transactional
+    protected List<Entitlement> entitleByProductsImpl(AutobindData data) throws EntitlementRefusedException {
```

Comment:
The Transactional annotation doesn't work with private methods, and I'm not sure we want to introduce the boilerplate it does to do the same work with private methods.